internal/api/v1/applications.go

```diff
@@ -149,7 +198,8 @@ func (hc ApplicationsController) Update(w http.ResponseWriter, r *http.Request)
 		return NewBadRequest("instances param should be integer equal or greater than zero")
 	}
 
-	err = app.Scale(r.Context(), updateRequest.Instances)
+	workload := application.NewWorkload(cluster, app)
+	err = workload.Scale(r.Context(), updateRequest.Instances)
```

Comment:
NOTE: We need to check if a `workload` actually exists before we try to scale it. E.g. 
- The user creates an application
- The user tries to scale it (before any staging has happened)
In this case we don't want to have a `500` error but rather let the user know they have to deploy first.